pkg/kanctl/profile_test.go

```diff
@@ -0,0 +1,103 @@
+package kanctl
+
+import (
+	"github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/secrets"
+	check "gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (k *KanctlTestSuite) TestConstructProfile(c *check.C) {
+	for _, tc := range []struct {
+		lp          *locationParams
+		secret      *v1.Secret
+		retCredType v1alpha1.CredentialType
+	}{
+		{
+			lp: &locationParams{
+				locationType: v1alpha1.LocationTypeS3Compliant,
+			},
+			secret: &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				StringData: map[string]string{
+					secrets.ConfigRole: "role",
+				},
+			},
+			retCredType: v1alpha1.CredentialTypeSecret,
+		},
+		{
+			lp: &locationParams{
+				locationType: v1alpha1.LocationTypeAzure,
+			},
+			secret: &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				StringData: map[string]string{
+					secrets.AzureStorageEnvironment: "env",
+				},
+			},
+			retCredType: v1alpha1.CredentialTypeSecret,
+		},
+		{
+			lp: &locationParams{
+				locationType: v1alpha1.LocationTypeAzure,
+			},
+			secret: &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				StringData: map[string]string{
+					secrets.AzureStorageEnvironment: "",
+				},
+			},
+			retCredType: v1alpha1.CredentialTypeKeyPair,
+		},
+		{
+			lp: &locationParams{
+				locationType: v1alpha1.LocationTypeAzure,
+			},
+			secret: &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				StringData: map[string]string{},
+			},
+			retCredType: v1alpha1.CredentialTypeKeyPair,
+		},
+		{
+			lp: &locationParams{
+				locationType: v1alpha1.LocationTypeGCS,
+			},
+			secret: &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+			},
+			retCredType: v1alpha1.CredentialTypeKeyPair,
+		},
+		{
+			lp: &locationParams{
+				locationType: v1alpha1.LocationTypeS3Compliant,
+			},
+			secret: &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+			},
+			retCredType: v1alpha1.CredentialTypeKeyPair,
+		},
+	} {
+		prof := constructProfile(tc.lp, tc.secret)
```

Comment:
Ah nice. I read the test file after adding my comment about testing in profile.go. Happy to see that you have already added the test 👍🏼 .